File: core/src/main/java/com/cangwang/core/ModuleEvent.java
package com.cangwang.core;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Created by zjl on 16/11/17.
 */
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.METHOD,ElementType.FIELD})
@Documented
public @interface ModuleEvent {
    Class<?> coreClientClass();
}

File: core/src/main/java/com/cangwang/core/ModuleEvent.kt
package com.cangwang.core

import java.lang.annotation.Documented
import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy
import kotlin.reflect.KClass

/**
 * Created by zjl on 16/11/17.
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER, AnnotationTarget.FIELD)
@Documented
annotation class ModuleEvent(val coreClientClass: KClass<*>, val single: Boolean = false)